src/nwb_conversion_tools/tools/roiextractors/roiextractors.py

```diff
@@ -503,23 +503,32 @@ def image_mask_iterator():
             )
 
             ps = image_segmentation.create_plane_segmentation(**input_kwargs)
-        # Fluorescence Traces:
+
+        # Fluorescence Traces - This should be a function on its own.
         if "Flourescence" not in ophys.data_interfaces:
             fluorescence = Fluorescence()
             ophys.add(fluorescence)
         else:
             fluorescence = ophys.data_interfaces["Fluorescence"]
-        roi_response_dict = segext_obj.get_traces_dict()
+
         roi_table_region = ps.create_roi_table_region(
             description=f"region for Imaging plane{plane_no_loop}",
             region=list(range(segext_obj.get_num_rois())),
         )
+
+        roi_response_dict = segext_obj.get_traces_dict()
+
         rate = np.nan if segext_obj.get_sampling_frequency() is None else segext_obj.get_sampling_frequency()
-        for i, j in roi_response_dict.items():
-            data = getattr(segext_obj, f"_roi_response_{i}")
-            if data is not None:
-                data = np.asarray(data)
-                trace_name = "RoiResponseSeries" if i == "raw" else i.capitalize()
+
+        # Filter empty data
+        roi_response_dict = {key: value for key, value in roi_response_dict.items() if value is not None}
+        for signal, response_series in roi_response_dict.items():
+
+            # all_data_is_zero = [value == 0 for value in list(response_series.flatten())]
+            all_data_is_zero = np.isclose(response_series, 0).all()
```

Comment:
I would rather this logical test be completely lazy in terms of memory - if the series is not being buffered, and it if it quite long, then the call to `np.isclose()` generates a full boolean array of the same shape. The array generated has, granted, `itemsize=1` for all elements, but that could still create instantaneous RAM usage of 50% of the original array amount if it uses `int16`.

The most efficient way I can think of would be

```python
any((x != 0 for x in np.ravel(response_series)))
```

`np.ravel` returns a 1D lazy view of the original array (does not create a copy of the shape) for easy iteration over all elements, the logical application is done via a generator so it technically only yields one logical element from the equality operation performed on a single element from the view at a time, and `any` even has a early return functionality where it stops iteration the first time it receives a `True` value from the iterable.

Reply:
This is clever, thanks.

Reply:
Done in the last commit.